### PR TITLE
                   Add ruff & docs

### PR DESCRIPTION
**Pull request by blueraft**

### CODE REVIEW

File: .github/workflows/actions.yaml [hunk 1]
```diff
@@ -13,9 +13,16 @@ jobs:
       run: |
         pip install --upgrade pip
         pip install -r requirements.txt --index-url https://gitlab.mpcdf.mpg.de/api/v4/projects/2187/packages/pypi/simple
```
Comment (hampusnasstrom) [inline .github/workflows/actions.yaml:15]: I was discussing with Pepe and he was having some issues with getting the right nomad-lab version. How are is the `requirements.txt` being generated here? I guess we should have a pipeline or script which generates this using the nomad-lab toml or requirements.

Comment (hampusnasstrom) [reply .github/workflows/actions.yaml:15]: Maybe this is not the right PR though, sorry. Just got this one linked ;)

Comment (blueraft) [reply .github/workflows/actions.yaml:15]: It's not a generated requirements file here, I just manually added ruff to that file. 

In this case the nomad-lab distribution is picked up from the gitlab pypi server, so the --index-url argument has to be passed to the pip install command for the correct version to be picked up. 

There's some discussion about specifying other pypi servers in the pyproject file itself [here](https://discuss.python.org/t/how-to-specify-extra-index-in-a-pyproject-toml-for-pip-and-pip-tools/23592), but it doesn't seem like that is something that's going to be supported.

File: mkdocs.yml [hunk 2]
```diff
@@ -0,0 +1,54 @@
+site_name: Documentation
+site_description: |
+  The documentation for NOMAD schema plugin example
+site_author: The NOMAD Authors
+repo_url: https://github.com/blueraft/nomad-schema-plugin-example
+
+nav:
+  - Home: index.md
+  - Explanation: explanation/explanation.md
+  - How-To Guides:
```
Comment (markus1978) [inline mkdocs.yml:10]: We should suggest typical how-tos:
- How to install this plugin
- How to use this plugin 
- How to contribute to this plugin
- How to contribute to the documentation

File: mkdocs.yml [hunk 3]
```diff
@@ -0,0 +1,54 @@
+site_name: Documentation
+site_description: |
+  The documentation for NOMAD schema plugin example
+site_author: The NOMAD Authors
+repo_url: https://github.com/blueraft/nomad-schema-plugin-example
+
+nav:
+  - Home: index.md
+  - Explanation: explanation/explanation.md
```
Comment (markus1978) [inline mkdocs.yml:9]: All the documentation pages should contain an attention box to emphasis that the documentation needs to be written.

**Review by markus1978 — COMMENTED** [review]
My comments should be pretty straight forward to implement. You can merge afterwards. After merging similar changes need to be implemented on the parser and normaliser example projects.

**Pull request merged**